docs/recipes/WritingTests.md

```diff
@@ -49,125 +76,26 @@ Then, add this to `scripts` in your `package.json`:
 
-```js
-import 'cross-fetch/polyfill'
-
-function fetchTodosRequest() {
-  return {
-    type: FETCH_TODOS_REQUEST
-  }
-}
-
-function fetchTodosSuccess(body) {
-  return {
-    type: FETCH_TODOS_SUCCESS,
-    body
-  }
-}
-
-function fetchTodosFailure(ex) {
-  return {
-    type: FETCH_TODOS_FAILURE,
-    ex
-  }
-}
-
-export function fetchTodos() {
-  return dispatch => {
-    dispatch(fetchTodosRequest())
-    return fetch('http://example.com/todos')
-      .then(res => res.json())
-      .then(body => dispatch(fetchTodosSuccess(body)))
-      .catch(ex => dispatch(fetchTodosFailure(ex)))
-  }
-}
-```
-
-can be tested like:
-
-```js
-import configureMockStore from 'redux-mock-store'
-import thunk from 'redux-thunk'
-import * as actions from '../../actions/TodoActions'
-import * as types from '../../constants/ActionTypes'
-import fetchMock from 'fetch-mock'
-import expect from 'expect' // You can use any testing library
-
-const middlewares = [thunk]
-const mockStore = configureMockStore(middlewares)
-
-describe('async actions', () => {
-  afterEach(() => {
-    fetchMock.restore()
-  })
-
-  it('creates FETCH_TODOS_SUCCESS when fetching todos has been done', () => {
-    fetchMock.getOnce('/todos', {
-      body: { todos: ['do something'] },
-      headers: { 'content-type': 'application/json' }
-    })
-
-    const expectedActions = [
-      { type: types.FETCH_TODOS_REQUEST },
-      { type: types.FETCH_TODOS_SUCCESS, body: { todos: ['do something'] } }
-    ]
-    const store = mockStore({ todos: [] })
-
-    return store.dispatch(actions.fetchTodos()).then(() => {
-      // return of async actions
-      expect(store.getActions()).toEqual(expectedActions)
-    })
-  })
-})
-```
+Our recommendation is to mock async requests at the `fetch/xhr` level using tools like [`msw`](https://mswjs.io/), [`miragejs`](https://miragejs.com/), [`jest-fetch-mock`](https://github.com/jefflau/jest-fetch-mock#readme), [`fetch-mock`](https://www.wheresrhys.co.uk/fetch-mock/), or similar. By mocking requests at this level, none of the thunk logic has to change in a test - the thunk still tries to make a "real" async request, it just gets intercepted. See the [components example](#example-1) for an example of testing a component which internally includes the behavior of async action creators.
 
 ### Reducers
 
-A reducer should return the new state after applying the action to the previous state, and that's the behavior tested below.
+A reducer should be a pure function that returns the new state after applying the action to the previous state. In the majority of cases, the reducer is an implementation detail that does not need explicit tests. However, if your reducer contains particularly complex logic that you would like the confidence of having unit tests for, reducers can be easily tested. As a reducer should always be a pure function, it is easy to write tests for - a given input should always provide a particular output.
```

Comment:
```suggestion
Reducers pure functions that return the new state after applying the action to the previous state. In the majority of cases, the reducer is an implementation detail that does not need explicit tests. However, if your reducer contains particularly complex logic that you would like the confidence of having unit tests for, reducers can be easily tested. 

Because reducers are pure functions, testing them should be straightforward.  Call the reducer with a specific input `state` and `action`, and assert that the result state matches expectations.
```